csrc/scheduler/pointwise_utils.cpp

```diff
@@ -150,16 +152,30 @@ bool DomainMap::areAllInputIdsMappedTo(TensorView* input_tv, TensorView* tv)
 bool DomainMap::eraseIfMapped(
     std::unordered_set<IterDomain*>& in_concrete_ids,
     IterDomain* out_id) const {
-  auto out_concrete_id =
-      ca_map_.getConcreteMappedID(out_id, IdMappingMode::PERMISSIVE);
-  auto in_concrete_id_iter = in_concrete_ids.find(out_concrete_id);
-  bool found_match = in_concrete_id_iter != in_concrete_ids.end();
-  if (found_match) {
-    in_concrete_ids.erase(in_concrete_id_iter);
+  if (getenv("OLD")) {
```

Comment:
When searching for a reference in these tensors:

```
Inputs:
  T0_g[ iS108{( ceilDiv(( ceilDiv(( ceilDiv(( 1024 * 1024 ), 4) ), 1) ), 128) )}, iS107{1}, iS105{4}, iS109{128} ], float
  T1_g[ iS84{( ceilDiv(( ceilDiv(( ceilDiv(( ( 1024 * 1024 ) * 4 ), 4) ), 1) ), 128) )}, iS83{1}, iS81{4}, iS85{128} ], float
Outputs:
  T7_g[ iblockIdx.x54{( ceilDiv(( ceilDiv(( ceilDiv(( ( 1024 * 1024 ) * 1 ), 4) ), 1) ), 128) )}, iUS53{1}, iV51{4}, ithreadIdx.x55{128} ] ca_pos( 2 ) produce_pos( 2 ), float
  T4_g[ iblockIdx.x96{( ceilDiv(( ceilDiv(( ceilDiv(( ( 1024 * 1024 ) * 4 ), 4) ), 1) ), 128) )}, iUS95{1}, iS93{4}, ithreadIdx.x97{128} ] ca_pos( 2 ) produce_pos( 4 ), float
```

`T7` is currently picked as the reference. This doesn't seem to make sense as `T7` has a broadcast domain (i.e., `1` in `1024 * 1024 ) * 1`), whereas `T1` has a corresponding concrete domain of extent 4 (`4` in `1024 * 1024 ) * 4`). `T7` doesn't seem to be a valid reference, but because we are using the PERMISSIVE map here, it is considered valid.

The change here is to use the exact map to find valid references. Since `in_concrete_ids` are permissive concrete IDs, valid references should be exactly mapped with them in pointwise fusions. In the example case of the above, this change picks `T4` as the reference as it has all three concrete domains.